temporian/core/compilation.py

```diff
@@ -24,57 +24,103 @@
-            as outputs.
+        fn: The function to compile. The function must take EventSetNodes as
+            arguments (and may have other arguments of arbitrary types) and
+            return EventSetNodes as outputs.
+        verbose: If >0, prints details about the execution on the standard error
+            output when the wrapped function is applied eagerly on EventSets.
+            The larger the number, the more information is displayed.
 
     Returns:
         The compiled function.
     """
 
-    @wraps(fn)
-    def wrapper(*args, **kwargs):
-        is_eager = None
-        args = list(args)
+    def _compile(fn: T) -> T:
+        @wraps(fn)
+        def wrapper(*args, **kwargs):
+            is_eager = None
+            args = list(args)
 
-        # EventSetNode -> EventSet mapping for eager evaluation
-        inputs_map = {}
+            # EventSetNode -> EventSet mapping for eager evaluation
+            inputs_map = {}
 
-        for i, arg in enumerate(args):
-            args[i], is_eager = _process_argument(
-                arg, is_eager=is_eager, inputs_map=inputs_map
-            )
+            for i, arg in enumerate(args):
+                args[i], is_eager = _process_argument(
+                    arg, is_eager=is_eager, inputs_map=inputs_map
+                )
 
-        for k, arg in kwargs.items():
-            kwargs[k], is_eager = _process_argument(
-                arg, is_eager=is_eager, inputs_map=inputs_map
-            )
+            for k, arg in kwargs.items():
+                kwargs[k], is_eager = _process_argument(
+                    arg, is_eager=is_eager, inputs_map=inputs_map
+                )
 
-        outputs = fn(*args, **kwargs)
+            outputs = fn(*args, **kwargs)
 
-        if is_eager is None:
-            raise ValueError(
-                "Cannot compile a function without EventSet or EventSetNode"
-                " argument."
-            )
-        elif is_eager:
-            from temporian.core.evaluation import run
+            if is_eager is None:
+                raise ValueError(
+                    "Cannot compile a function without EventSet or EventSetNode"
+                    " argument."
+                )
+            elif is_eager:
+                from temporian.core.evaluation import run
+
+                return run(query=outputs, input=inputs_map, verbose=verbose)
+
+            return outputs
+
+        wrapper.is_tp_compiled = True
 
-            return run(query=outputs, input=inputs_map)
+        return wrapper
 
-        return outputs
+    # Function is being called as a decorator
+    if fn is not None:
+        # Set default values for kwargs
+        verbose = 0
 
-    wrapper.is_tp_compiled = True
+        return _compile(fn)
 
-    return wrapper
+    # Else the function is being called as a function, so we return a decorator
+    # that will receive the function to compile.
+    verbose = verbose or 0
```

Comment:
I would make the  value of verbose be 0 in the function header is possible.

Reply:
this logic was leftover from a previous worse implementation, good catch :)